peachjam/views/generic_views.py

```diff
@@ -45,6 +45,8 @@ def get_base_queryset(self):
         return qs.preferred_language(self.get_language())
 
     def get_queryset(self):
+        if self.form.data.get("alphabet"):
+            return self.form.filter_queryset(self.get_base_queryset()).order_by("title")
```

Comment:
You're repeating code here. It's better to do what is common in both cases once -- get the queryset and filter it -- and then apply the appropriate ordering.

Something like:

```python
qs = self.form.filter_queryset(self.get_base_queryset()).order_by("-date")
if ...:
  qs = qs.order_by("title")
return qs
```

It would also make sense to me for the form itself to apply the ordering in `filter_queryset`. The form has the knowledge of the form fields, etc.

Reply:
See https://en.wikipedia.org/wiki/Don%27t_repeat_yourself